platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java

```diff
@@ -1058,309 +772,307 @@ private void setIsConsensusTrue(EventImpl event, RoundInfo receivedRoundInfo) {
+                            weight += getWeight(m3);
                         }
                     }
+                    if (Threshold.SUPER_MAJORITY.isSatisfiedBy(weight, totalWeight)) { // strongly see supermajority of
+                        // intermediates
+                        x.setStronglySeeP(mm, st);
+                    } else {
+                        x.setStronglySeeP(mm, null);
+                    }
                 }
             }
-            result = x.getStronglySeeP(m);
         }
-        t = System.nanoTime() - t; // nanoseconds spent doing the dot product
-        consensusMetrics.dotProductTime(t);
-        return result;
+        return x.getStronglySeeP((int) m);
     }
 
     /**
-     * The round-created for event x (first round is 1), or 0 if x is null (function from SWIRLDS-TR-2020-01).
-     * It also stores the round number with x.setRoundCreated().
-     * This result is memoized.
+     * The round-created for event x (first round is 1), or 0 if x is null (function from
+     * SWIRLDS-TR-2020-01). It also stores the round number with x.setRoundCreated(). This result is
+     * memoized.
      *
-     * If the event has a hash in the hash lists given to the ConsensusImpl constructor, then the roundCreated is set
-     * to that round number, rather than calculating it from the parents.
+     * <p>If the event has a hash in the hash lists given to the ConsensusImpl constructor, then the
+     * roundCreated is set to that round number, rather than calculating it from the parents.
      *
-     * If a parent has a round of -1, that is treated as negative infinity. So if all parents are -1, then this one
-     * will also be -1.
+     * <p>If a parent has a round of -1, that is treated as negative infinity. So if all parents are
+     * -1, then this one will also be -1.
      *
-     * @param x
-     * 		the event being queried
+     * @param x the event being queried
      * @return the round-created for event x, or 0 if x is null
      */
-    private long round(EventImpl x) {
-        int numMembers = getAddressBook().getSize(); // number of members that are voting, with ID 0 to numMembers-1
-        EventImpl op, sp; // other parent, self parent
-        long rop, rsp, weight; // roundCreated of other parent, roundCerated of self parent, sum of weight involved
-
+    private long round(@Nullable final EventImpl x) {
+        //
+        // If an event is missing (null), it must be ancient. ancient events have a round of
+        // -infinity
+        //
         if (x == null) {
-            return 0;
+            return ConsensusConstants.ROUND_NEGATIVE_INFINITY;
         }
-        if (x.getRoundCreated() > 0) {
+
+        //
+        // Is the round already memoized? If it is, return the memoized value
+        //
+        if (x.getRoundCreated() >= ConsensusConstants.ROUND_NEGATIVE_INFINITY) {
```

Comment:
I'm a cycle counter, it would reduce a method call to get the value and store it in a local variable instead of getting it twice.

Reply:
This paradigm is something Leemon insists upon, so nothing I can do here